File: spec/DigitsGameSpec.js
describe("DigitsGame", function() {
	var game;
	var statements;

	beforeEach(function(){
		spyOn(DigitsUtil, 'random').andReturn(1234);
		game = new DigitsGame();
		statements = {
			win: "win",
			going: "try again",
			lose: "lose"
		}
	});	

	it("should return 'win' when input a right number <= six times", function(){
		expect(game.run(1234)).toEqual(statements.win);	
		expect(game.runAgain(1345, 1456, 1234)).toEqual(statements.win);	
		expect(game.runAgain(1345, 1456, 1278, 1289, 1234)).toEqual(statements.win);
		expect(game.runAgain(1345, 1456, 1234, 1289, 2345, 1234)).toEqual(statements.win);
	});	

	it("should return 'try again' when input a wrong number < six times",function(){
		expect(game.run(9876)).toEqual(statements.going);
		expect(game.runAgain(9876, 7659)).toEqual(statements.going);
		expect(game.runAgain(9876, 1456, 1235, 1289, 2345)).toEqual(statements.going);
	})


	it("should return 'lose' when input a wrong number >= six times",function(){
		expect(game.run(9876, 1456, 1235, 1289, 2345, 9567)).toEqual(statements.lose);
		expect(game.runAgain(9876, 7659, 1235, 1289, 2345, 9567, 8475)).toEqual(statements.lose);
	})

	it("should return 'lose' even innput a right number > six times", function(){
		expect(game.runAgain(9876, 7659,1235, 1289, 2345, 9567, 1234)).toEqual(statements.lose);
	})

});

File: spec/digits_game_spec.rb
require 'digits_game'
require 'spec_helper'

DigitsGame.class_eval do
  def init
    @target_number = 1234
  end
end

describe DigitsGame do
  before(:each) do
    @game = DigitsGame.new
    @game.init
  end

  it "should return winner when the target number guessed" do
    @game.run(1234).should == "winner"
  end

  it "should return try again when guessed wrong and guess count < 6" do
    @game.run(4321, 1235, 3214).should == "try again"
  end

  it "should return lose when guessed over 6" do
    @game.run(4321, 1235, 3214, 5678, 8904).should == "try again"
    @game.run(4321, 1235, 3214, 5678, 8904, 1356).should == "lose"
    @game.run(4321, 1235, 3214, 5678, 8904, 8035, 4568).should == "lose"
  end

end